src/program/UplcProgram.js

```diff
@@ -24,7 +26,7 @@ import { UplcCall, UplcConst, UplcForce, UplcReader } from "../terms/index.js"
  * @typedef {{
  *   root: UplcTerm
  *   ir: Option<string>
- *   eval(args: Option<UplcValue[]>, costModelParams?: number[]): CekResult
+ *   eval(args: Option<UplcValue[]>, costModelParams?: number[], logOptions?: Option<Partial<UplcLoggingI>>): CekResult
```

Comment:
In my experience, if there is more than 1 optional arg it is better to place them into a `props` or `config` or `options` object

Reply:
I was kind of thinking it would make sense to make it non-optional but not mess up the interface during transition phase.

WDYT about moving it next to `args` and making it required?

Reply:
I'll bump the minor version if we add breaking changes, don't worry.

Maybe it makes sense for the default setting not to use `console.log` at all, and put all logs in the return value. When a logger is specified the CekMachine then uses that logger in addition to placing all logs in the return value.

What do you think?

Reply:
I think that gives us good flexibility.  A logging object can do things its own way, while the CekMachine's `result.logs` can contain the logs of a single script.  

Earlier versions of transaction execution used to throw errors containing all those logs, so I'm imagining that this approach will help some with continuity.  Perhaps a `throwFullScriptLogs` option (default true) would tie up that story elegantly?

Reply:
`throwFullScriptLogs` as part of the `TxBUilder` or elsewhere is a great idea. Although it doesn't have to be implemented as part of this set of PRs. 

(None of the stuff I'm doing is relying on how or what errors are being thrown or how messages are being logged)